### PR TITLE
LightGBM parameter changes to match Python implementation results

### LINKED ISSUE #654: Default value of sigmoid parameter for LightGBM
[The default value of the `sigmoid` parameter for LightGBM is 1](https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.rst#sigmoid), however in the ML.Net wrapper it is hardwired to 0.5:
https://github.com/dotnet/machinelearning/blob/f9d3973a056ad26bc6cc15c2d7a09f8ae47e30da/src/Microsoft.ML.LightGBM/LightGbmArguments.cs#L404

Why is this the case? Shouldn't this parameter either be 1 or exposed on the ML.Net API instead?

### LINKED ISSUE #6062: LightGBM should use default seed if not set explicitly
**System Information (please complete the following information):**
 - OS & Version: Windows 11
 - ML.NET Version: ML.NT 1.6.0
 - .NET Version: .NET 5.0

**Describe the bug**
LightGBM must get random seed when called, or it will set a set a random seed from MLContext.
However, LightGBM has a default value for seeds. If we set this random_seed, then other seeds will get overridden. ML.NET interface does not provide option to set the other seeds explicitly.

Links:
https://lightgbm.readthedocs.io/en/latest/Parameters.html#seed
https://github.com/dotnet/machinelearning/blob/b3a3c665aa3d986ffd06a221f07b46f589e45c75/src/Microsoft.ML.LightGbm/LightGbmTrainerBase.cs#L230

If some features which use randomity are used then this may cause issues:
-not being able to reproduce results after running Python. 
-not able to reproduce results in ML.NET due to MLContex random generator may have had a different number of calls already.

**Expected behavior**
Seed should not be set for LightGBM. 
If not possible, then option to set the specific seeds ( data_random_seed, feature_fraction_seed ...) would be useful.

**Possible fix*
Do not set res{"seed"] here if Seed == null 
[res["seed"] = (Seed.HasValue) ? Seed : host.Rand.Next();](https://github.com/dotnet/machinelearning/blob/b3a3c665aa3d986ffd06a221f07b46f589e45c75/src/Microsoft.ML.LightGbm/LightGbmTrainerBase.cs#L267)

**Additional context**
Related to this discussion about not being able to reproduce LightGBM results in ML.NET from Python
https://github.com/microsoft/FLAML/discussions/409#discussioncomment-2058431

### LINKED ISSUE #6063: Q: Where is num_iterations set for LightGBM binary?
I am trying to match model performance for LightGBM in .NET and Python. I am having some challenges ( discussion :  https://github.com/microsoft/FLAML/discussions/409#discussioncomment-2058431 )

Testing pipeline with simple Titanic dataset simplified sample https://github.com/torronen/lightgbm-comparison

I would like to understand better how values are sent for LightGBM library compiled from C++.
I have trouble finding how NumberOfIterations is mapped. 

The below method would rename it to number_of_iterations but it is not a valid alias for number of trees per link to C++ source below. I did not find "NumberOf" => "num" replace either.
https://github.com/dotnet/machinelearning/blob/212b5dfc3ba9341ea73867cdc495dbcada8ab660/src/Microsoft.ML.LightGbm/WrappedLightGbmInterface.cs#L258-L282 

I think it should be here:
https://github.com/dotnet/machinelearning/blob/b3a3c665aa3d986ffd06a221f07b46f589e45c75/src/Microsoft.ML.LightGbm/LightGbmTrainerBase.cs#L50-L61

I can find NumBoostRounds (num_boost_rounds is a valid alias) here, but could someone explain what does this annotation mean? There is another class for sweepable ranges. Does this get somehow transferred to LightGBM binary? Where does this suggested sweepable range get used?
https://github.com/dotnet/machinelearning/blob/8eac93a0f999acb144478f1b39a321128861acc3/src/Microsoft.ML.LightGbm/LightGbmTrainerBase.cs#L77

Finally, those values which have dependant default values are set here:
https://github.com/dotnet/machinelearning/blob/8eac93a0f999acb144478f1b39a321128861acc3/src/Microsoft.ML.LightGbm/LightGbmTrainerBase.cs#L431-L433

Explanation of some name mapping + links can be found from this Github issue. Although snippets point to old versions, main branch seems not to have changes
https://github.com/dotnet/machinelearning/issues/5451#issuecomment-714664503

**Links to LightGBM aliases:** 
C++ source
https://github.com/microsoft/LightGBM/blob/cf38071b6ac7e238b1654286dbb5a9a98b2c320b/src/io/config_auto.cpp
(more clear in R package):
https://github.com/microsoft/LightGBM/blob/c84b330a30c0b6b8b71ec5d2bffab05928a715d0/R-package/R/aliases.R#L108
Aliases in documentation
https://lightgbm.readthedocs.io/en/latest/Parameters.html#num_iterations
Same case in 2.3.1
https://github.com/microsoft/LightGBM/blob/v2.3.1/src/io/config_auto.cpp

### PR DESCRIPTION
**Pull request by torronen**
Suggested changes for LightGBM results through ML.NET similar as through Python:
* keep LightGBM default seed if seed has not been set
* add mapping from NumberOfIterations to num_iterations
* add NumberOfIterations to parameters array for LightGBM
* change sigmoid default value to match LightGBM
* Default Evaluation Metric to None per LightGBM default

Project that can be used for comparison between LightGBM in Python and Microsoft.ML.LightGBM and also compare ModelBuilder with python-FLAML: https://github.com/torronen/lightgbm-comparison

Rationale: https://github.com/microsoft/FLAML/discussions/409#discussioncomment-2058431
Reasons for changes explained in the issues:
* fixes #6063 
* fixes #6062 
* fixes #654

I suggest the results should be equal through Python and ML.NET so that developers can discuss and share best practices about hyperparameters. Also, it enables to use tuning from Python. 

Sigmoid value change has been propose before but was not implemented.
It may need more consideration:  https://github.com/dotnet/machinelearning/pull/667

PR is for comments and discussions for now. Results are not yet equal through ML.Net and Python.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/machinelearning?pullRequest=6064) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/machinelearning?pullRequest=6064) before we can accept your contribution.<br/><br/>:x: torronen [sign now](https://cla.dotnetfoundation.org/dotnet/machinelearning?pullRequest=6064)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/machinelearning?pullRequest=6064) it.</sub>

**Comment by torronen** [comment]
lgbm1
lgbm2

**Comment by torronen** [comment]
Is there any reason to use aliases? If not, I suggest we update

- main_split_gain=main_split_to_gain  
- min_sum_hessian_in_leaf=min_child_weight   
- bagging_freq = subsample_freq
- bagging_fraction = subsample 
- lambda_l2 =reg_lambda
- lambda_l1 =reg_alpha
- boosting =boosting_type
- verbosity =verbose
- unbalanced_sets = is_unbalance

**Comment by torronen** [comment]
Iterations actually seem to be ran inside .NET, so it does not need to be passed anywhere.
https://github.com/dotnet/machinelearning/blob/8eac93a0f999acb144478f1b39a321128861acc3/src/Microsoft.ML.LightGbm/WrappedLightGbmTraining.cs#L19

I will close this PR as it is better not to update the defaults if it does not provide any better performance. Changing default may be a breaking change for some developers.

However, for some reason Python seems to provide better speed and accuracy for LightGBM (and therefore, for many applications) at least on a few datasets I've tried.
